internal/service/advancedcluster/resource_advanced_cluster_migration_test.go

```diff
@@ -217,11 +224,12 @@ func TestMigAdvancedCluster_partialAdvancedConf(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "bi_connector_config.0.read_preference", "secondary"),
 				),
 			},
+			mig.TestStepCheckEmptyPlan(configUpdated),
```

Comment:
could we move this check above the step that actually applies the updated config? Essentially ensure that without applying the new config, an non empty plan is encountered.

Reply:
my bad, I believe the existing test covers it correctly: 

- first step defines the config with int using latest provider version (schema with int)
- second step continues using int but uses local built provider that has float schema, and verifies empty plan before applying

I will let you confirm but I believe we are good

Reply:
Yes, that is correct 👍
Added the `mig.TestStepCheckEmptyPlan(configUpdated)` L227 to check the updated config doesn't have any plan changes too as it was missing